assets/shaders/Scatter.glsl

```diff
@@ -13,7 +13,7 @@ float Schlick(const float cosine, const float refractionIndex)
 RayPayload ScatterLambertian(const Material m, const vec3 direction, const vec3 normal, const vec2 texCoord, const float t, inout uint seed)
 {
 	const bool isScattered = dot(direction, normal) < 0;
-	const vec4 texColor = m.DiffuseTextureId >= 0 ? texture(TextureSamplers[m.DiffuseTextureId], texCoord) : vec4(1);
+	const vec4 texColor = m.DiffuseTextureId >= 0 ? texture(TextureSamplers[nonuniformEXT(m.DiffuseTextureId)], texCoord) : vec4(1);
```

Comment:
Shouldn't the top of the file ideally have a `#extension GL_EXT_nonuniform_qualifier: require` ?

Reply:
Probably, I will include the extensions.